### PR TITLE
Add feature and product maturity levels info

### PR DESCRIPTION
**Pull request by adamkvitek**
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [x] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [ ] Redirects are added if the PR changes page URLs

### CODE REVIEW

**Review by milanwilliams — APPROVED** [review]

File: docs/supported-languages.md [hunk 1]
```diff
@@ -142,6 +143,10 @@ Visit the cheat sheet generation script and associated semgrep-core test files t
 * [Generation script](https://github.com/returntocorp/semgrep/blob/develop/scripts/generate_cheatsheet.py)
 * [`semgrep-core` test files](https://github.com/returntocorp/semgrep/tree/develop/semgrep-core/tests)
 
+:::info Feature and product maturity levels
+The detailed specifications given above apply only to language support. Language maturity levels differ from feature and product maturity levels. Unlike supported languages, features, and products marked here as experimental (alpha), beta or GA carry meaning used regularly in software terminology as a part of the software release life cycle. For a standard definition of these terms, see for example [Software release life cycle](https://en.wikipedia.org/wiki/Software_release_life_cycle).
```
Comment (s-santillan) [inline docs/supported-languages.md:147]:
```suggestion
* The detailed specifications given above apply only to language support. Language maturity levels differ from feature and product maturity levels.
* Semgrep features and products documented as experimental (alpha), beta, or GA generally follow the definitions in a [Software release life cycle](https://en.wikipedia.org/wiki/Software_release_life_cycle).
```

Minor and can be ignored. Just shortened it. "Carry meaning" is also not a common English turn of phrase.

Comment (pabloest) [inline docs/supported-languages.md:147]:
```suggestion
The detailed specifications given above apply only to language support. Language maturity levels differ from feature and product maturity levels. Unlike supported languages, features, and products marked here as experimental, beta or GA carry meaning used regularly in software terminology as a part of the software release life cycle. For a standard definition of these terms, see for example [Software release life cycle](https://en.wikipedia.org/wiki/Software_release_life_cycle).
```

**Review by s-santillan — APPROVED** [review]
Only 1 minor comment.

**Review by pabloest — APPROVED** [review]

File: docs/supported-languages.md (hunk 1 above)
Comment (adamkvitek) [reply docs/supported-languages.md:147]: I will remove alpha after committing Sara's suggestion. Thak you!

Comment (adamkvitek) [reply docs/supported-languages.md:147]: Thank you! Great edit. :-)

File: docs/supported-languages.md [hunk 2]
```diff
@@ -140,7 +141,12 @@ The following **thresholds** define each maturity level:
 Visit the cheat sheet generation script and associated semgrep-core test files to learn more about each feature:
 
 * [Generation script](https://github.com/returntocorp/semgrep/blob/develop/scripts/generate_cheatsheet.py)
-* [`semgrep-core` test files](https://github.com/returntocorp/semgrep/tree/develop/semgrep-core/tests)
+* [`semgrep-core` test files](https://github.com/returntocorp/semgrep/tree/develop/tests)
+
+:::info Feature and product maturity levels
+* The detailed specifications given above apply only to language support. Language maturity levels differ from feature and product maturity levels.
+* Semgrep features and products documented as experimental (alpha), beta, or GA generally follow the definitions in a [Software release life cycle](https://en.wikipedia.org/wiki/Software_release_life_cycle).
```
Comment (adamkvitek) [inline docs/supported-languages.md:148]:
```suggestion
* Semgrep features and products documented as experimental, beta, or GA generally follow the definitions in a [Software release life cycle](https://en.wikipedia.org/wiki/Software_release_life_cycle).
```

**Pull request merged**